File: 2582-pass-the-pillow/2582-pass-the-pillow.cpp
class Solution {
public:
    int passThePillow(int n, int time) {
         int fullrounds = time / (n - 1); 
         int ans = 0; 
        if (fullrounds % 2 == 0) {
           
            ans = 1 + (time % (n - 1));
        } else {
            
            ans = n - (time % (n - 1));
        }

        return ans; 
    }
    };

 

File: 2582-pass-the-pillow/2582-pass-the-pillow.py
class Solution:
    def passThePillow(self, n: int, time: int) -> int:
        if n == 1:
            return 1
        cycle = 2 * (n - 1)
        effective_time = time % cycle
        if effective_time <= n - 1:
            return effective_time + 1
        else:
            return 2 * (n - 1) - effective_time + 1